src/v/raft/heartbeat_manager.cc

```diff
@@ -204,6 +212,8 @@ void heartbeat_manager::process_reply(
                 vlog(hbeatlog.error, "cannot find consensus group:{}", g);
                 continue;
             }
+            (*it)->suppress_heartbeats(
```

Comment:
supress_heartbeats should maybe be named `drop_network_heartbeats_immediately()`  or something like that, because this is an *specific* kind of supression. 

Alternatively, if we have multiple supression types, then it should have an explicit enum 

`supress_heartbeat(supression_types::drop_network_request,  arg2, arg3)`

thoughts?

Reply:
I would prefer two different methods, I am just wondering about the name, the heartbeat is suppressed because another heartbeat request is in progress, maybe just `start_heartbeat_request(raft::vnode)`.

Reply:
Why not an enum? 

I don’t understand how `start_heartbeat_request` is related at all to suppressing? 

What are the 2 types of suppression ?

Reply:
It is always other request being in progress. It is always append entries request. Can be either heartbeat or data.

Reply:
Sure, but we are talking about the name of a method that *drops* RPC’s. 

So i think the proposals should be around that no? I’m confused otherwise?

Reply:
I went over this once again, and I think that there is always one kind of suppression i.e. we suppress sending heartbeat when other request is in progress. We use follower `sequence_number` to prevent race conditions. We do not need to recognize the type of suppression. If you think it will be more readable I can rename `suppress_heartbeats(...)` -> `drop_heartbeats(...)`

Reply:
there is exactly one reason to drop heartbeats and it has to do with our implementation splitting heartbeats and append_entries due to performance optimizations. 

thanks to @mmaslankaprv for live walking me through this.

Reply:
let's add a heartbeat to the reciever RPC layer as well.

Reply:
let's add a comment abou tthe implementation on the conensus.h supress_heartbeat

Reply:
other than that LGTM